src/components/common/ui/form/input-mui/Input.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import {
+  FormControl,
+  FormHelperText,
+  InputLabel,
+  OutlinedInput,
+} from '@mui/material';
+import { SxProps, Theme } from '@mui/material/styles';
+import { useField } from 'formik';
+
+import {
+  getLeftIcon,
+  getRightIcon,
+  getState,
+} from '@/components/common/ui/form/input-mui/util';
+import mergeSx from '@/lib/utils/MergeSxStylesUtil';
+
+import * as styles from './Input.styles';
+
+export enum InputState {
+  DISABLED = 'disabled',
+  DEFAULT = 'default',
+  SUCCESS = 'success',
+  ERROR = 'error',
+}
+
+export enum InputSize {
+  LARGE = 'large',
+  MEDIUM = 'medium',
+}
+
+export enum InputType {
+  DEFAULT = 'text',
+  PASSWORD = 'password',
+  SEARCH = 'search',
+}
+
+interface InputProps
+  extends Omit<React.ComponentPropsWithoutRef<'input'>, 'size'> {
+  label?: string;
+  placeholder?: string;
+  size?: InputSize;
+  type?: InputType;
+  isSuccessOnDefault?: boolean;
+  defaultRemark?: string;
+  showRemark?: boolean;
+  sx?: SxProps<Theme>;
+  onDeterredChange?: () => void;
+  isImmutable?: boolean;
+}
+
+const MAX_LENGTH = 2000;
+
+const Input: React.FC<InputProps> = ({
+  label,
+  placeholder,
+  size = InputSize.LARGE,
+  type = InputType.DEFAULT,
+  isSuccessOnDefault = false,
+  defaultRemark,
+  showRemark = true,
+  sx,
+  onDeterredChange,
+  disabled = false,
+  ...rest
+}) => {
+  const [field, { touched, error }, { setTouched, setValue }] = useField(
+    rest.name,
```

Comment:
get `name` with destruction, not with `rest`. you might not need to pass to input, you can just extract `value`, and create handlers for input